src/Nethermind/Nethermind.AccountAbstraction/Executor/UserOperationSimulator.cs

```diff
@@ -98,13 +98,13 @@ public class UserOperationSimulator : IUserOperationSimulator
                 }
             }
 
-            IReleaseSpec currentSpec = _specProvider.GetSpec(parent.Number + 1);
+            IReleaseSpec specfor1559 = _specProvider.GetSpecFor1559(parent.Number + 1);
             ReadOnlyTxProcessingEnv txProcessingEnv = _readOnlyTxProcessingEnvFactory.Create();
             ITransactionProcessor transactionProcessor = txProcessingEnv.Build(_stateProvider.StateRoot);
 
             // wrap userOp into a tx calling the simulateWallet function off-chain from zero-address (look at EntryPoint.sol for more context)
             Transaction simulateValidationTransaction =
-                BuildSimulateValidationTransaction(userOperation, parent, currentSpec);
+                BuildSimulateValidationTransaction(userOperation, parent, specfor1559);
```

Comment:
Don't we need a proper ReleaseSpec here with timestamp forks too? What if `Eip1559BaseFeeMinValue` changed?